hieradata/class/backend.yaml

```diff
@@ -36,6 +36,7 @@ govuk::node::s_base::apps:
   - release
   - search_admin
   - service_manual_publisher
+  - share-sale-publisher
```

Comment:
This needs to match the Puppet class name, eg `share_sale_publisher` (you're using hyphens rather than underscores)